### ISSUE TITLE
Allow scaling a path-following CSGPolygon

### ISSUE THREAD

**Issue by kb173** [godotengine/godot-proposals#2412]
### Describe the project you are working on

A landscape visualization which includes dynamically generated streets from geodata. For these, we have a CSGPolygon with a PoolVector2Array polygon, which is extruded along the paths representing the streets.

### Describe the problem or limitation you are having in your project

We would like to generically resize that polygon without having to know its exact points, so that we can easily adapt any street profile to the exact width at that segment. For example, we would pre-define a street profile at a width of 2m and then dynamically adapt that width to the actual width at a given position. I had expected to be able to do this by setting the 'Scale' parameter in its Transform.

However, the scale entered in the CSGPolygon's Transform is only visible until a different node is selected; then is resets (only visually - the different value is still in the Transform). For the short moment where the scale is visible, it doesn't actually scale the polygon, but the whole mesh extruded along the path. This is the result I would expect when scaling the path, not the polygon.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Scaling the CSGPolygon by setting its 'Transform -> Scale' parameter should scale the polygon itself, so that a line which is extruded from it becomes thicker/thinner.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

There is no need for new editor tabs or code functions - the existing Scale from the Transform which CSGPolygon inherits from would be used as described above.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

With a scale of 1, everything will remain the same. The behavior for non-1 scale is changed - in a way, this breaks compatibility for projects which rely on the scale not affecting the CSGPolygon, but the current behavior seems more like a bug to me?

### Is there a reason why this should be core and not an add-on in the asset library?

I intuitively expected scaling the CSGPolygon to have the result described above; I assume that it might be similar for others. Additionally, as I said above, the current behavior feels like a bug due to the short moment in which the scale is visible (in an unexpected way). _Some_ type of change is required to clear up confusion (e.g. an editor warning when scaling a CSGPolygon, if this enhancement is not implemented).

(This supercedes https://github.com/godotengine/godot/issues/32171)

**Comment by MathiasBaumgartinger** [comment]
I heavily support this problem. It feels highly unintuitive the way it is now.

**Comment by Calinou** [comment]
Thinking about it, isn't this a duplicate of https://github.com/godotengine/godot-proposals/issues/1062 ?

**Comment by Zireael07** [comment]
Yep, looks like a dupe.

**Comment by kb173** [comment]
Oh true, it's similar, but not quite the same I think -- the other proposal is about being able to have a _different_ CSGPolygon for the start and the end, while this proposal calls for being able to use the `Transform`'s Scale property for scaling one CSGPolygon involved in a PathFollow. (I suppose that the other proposal might benefit from the implementation of this one, since it mentions scaling, but in that case it expands on this proposal.)

**Comment by stebulba** [comment]
> (I suppose that the other proposal might benefit from the implementation of this one, since it mentions scaling, but in that case it expands on this proposal.)

No I included a scaling mode in my code: https://github.com/godotengine/godot/pull/41499

> The interpolation work using a Curve who interpolate between two polygon array. If both polygons are equal, a single scaling mode is active.

By default, if you did't modify the second array, the scaling mode is activate. If you modify the second array, the expand mode is activate. I like "expand" to call this mode :)

Remember, that if you want scaling a perfect square, you have to set up yours arrays with the position 0, 0 as center. By default the path polygon make a rectangle in positive position.

**Comment by kb173** [comment]
> @kb173 I made a new pull request. [godotengine/godot#53195](https://github.com/godotengine/godot/pull/53195) Does this solve your limitation ?

@stebulba Yes, this looks amazing! The x-axis scaling you show off in the first video is exactly what we need, given that we're working with streets. Thank you for your work!
